A51-xds-custom-backend-metrics.md

```diff
@@ -0,0 +1,382 @@
+and the subscribing LB polic(ies) will not receive any callbacks.  
+However, the client will record a channel trace event indicating that this has happened.  
+It will also log a message at priority ERROR.
+
+If the StreamCoreMetrics call returns any other status, the client will retry the call.  
+To avoid hammering a server that may be experiencing problems, the client will use exponential backoff between attempts.  
+When the client receives a message from the server on a given call, the backoff state is reset, 
+so the next attempt will occur immediately, but any subsequent attempt will be subject to exponential backoff.
+
+When the client subchannel is shutting down or when the backend sends a GOAWAY, 
+the client will cancel the StreamCoreMetrics call.  
+There is no need to wait for the final status from the server in this case.
+
+## Use Cases
+
+Enumerating specific metrics types and their reporting mechanisms, we have:
+
+1. Collect Utilization Metrics using Out-of-Band Reporting
+
+It is popular for a gRPC powered environment to be running its own intra-locality WRR LB policy plugin 
+at the client side, which will need to access the backend utilization metrics feedback that is further 
+translated to subchannel weights for the scheduling algorithm to function, 
+e.g. CPU, queue depth. Observing these metrics is fulfilled by the gRPC custom backend metrics feature.
+
+2. Collect Utilization Metrics using Per-Query Reporting 
+
+In a high QPS environment, reporting request cost using per-query is desired. 
+Compared with 1, the advantage of per-query is that the utilization metrics can be timely aligned with the requests. 
+It also communicates the metrics data rapidly and saves cost of the idle connection compared with OOB reporting.
+
+3. Collect Request Cost Metrics in Per-Query Reporting
+
+Measuring the query cost from the client perspective and aggregate at a global load balancer is 
+another proven traffic distribution strategy. 
+A global LB can balance live user traffic between clusters, so that it can match user demand to 
+available service capacity, and handle service failures in a fashion that is transparent to the users. 
+gRPC’s native solution of per-query metrics reporting supports this use case.
+
+4. Collect Request Cost Metrics in Out-of-Band Reporting 
+
+No use case.  it is also technically more complicated than per-query reporting of request costs, 
+because in OOB we need to identify which client the request cost reports is generated from and associate 
+it with the corresponding OOB stream in order to deliver the report accurately. 
+So generating the request cost report requires extra processing for each distinct client connection in the OOB scenario.
+This scenario is not supported until there is use case in the future.
+
+## Rationale
+
+## Implementation
+
+This will be implemented simultaneously in Java by @YifeiZhuang, in
```

Comment:
I don't think we need to say who's doing the implementation.  This line can probably just be removed.